File: hash map/contains_duplicate.cpp
// C++ version of solution to Contains Duplicate II (python)

#include <unordered_map>

class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
            unordered_map<int, int> mapIndex;
            // traverse through vector
            for(int i=0;i<nums.size();i++){
                if(mapIndex.find(nums[i]) != mapIndex.end()) // not equal to means nums[i] was found
                {
                    int sameIndex = mapIndex[nums[i]]; // get the index (value) by using its key (nums[i])
                    if(i - sameIndex <= k){
                        return true;
                    }
                }
                mapIndex[nums[i]] = i; // if current element not in list
            }    
            return false;
    }
};

File: hash map/contains_duplicate.py
class Solution:
    def hasDuplicate(self, nums: List[int])->List[int]:
        mapList = {}
        for i in range(0, len(nums)):
            if nums[i] in mapList:
                if i != mapList[nums[i]]:
                    return True
            else:
                mapList[nums[i]] = i
        return False
